src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -48,24 +53,88 @@
 @Log4j2
 public class HybridQueryPhaseSearcher extends QueryPhaseSearcherWrapper {
 
+    final static int MAX_NESTED_SUBQUERY_LIMIT = 20;
+
     public HybridQueryPhaseSearcher() {
         super();
     }
 
     public boolean searchWith(
         final SearchContext searchContext,
         final ContextIndexSearcher searcher,
-        final Query query,
+        Query query,
         final LinkedList<QueryCollectorContext> collectors,
         final boolean hasFilterCollector,
         final boolean hasTimeout
     ) throws IOException {
-        if (query instanceof HybridQuery) {
+        if (isHybridQuery(query, searchContext)) {
+            query = extractHybridQuery(searchContext, query);
             return searchWithCollector(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
         }
+        validateQuery(query);
```

Comment:
not able to understand why we need this check?

Reply:
this is needed to track cases when hybrid query is wrapped into other queries, for now we're checking only bool query. Something like the query below:

```
{
    "query": {
        "bool": {
            "should": [
                {
                    "hybrid": {
                        "queries": [
                            {
                                "nested": {
                                    "path": "user",
                                    "query": {
                                        "match": {
                                            "user.name": "john"
                                        }
                                    }
                                }
                            },
                            {
                                "match": {
                                    "name": {
                                        "query": "team"
                                    }
                                }
                            }
                        ]
                    }
                },
                {
                    "match": {
                        "name": {
                            "query": "university"
                        }
                    }
                }
            ]
        }
    }
}
```

Reply:
what I am not able to understand is what we are validating and if validation fails what we are doing.

Reply:
the logic if following:
check if query is hybrid
if true - extract hybrid query if that's needed (case reported in other github issue, when it's wrapped into Bool query behind the scene by opensearch core code)
else (query isn't hybrid) - validate if hybrid is part of other compound query
     if true - throw error as we allow hybrid only as a top level query
     else  - execute query

Reply:
I think I got it what you are trying to do here. can we add documentation on top of `validateQuery` function around what kind of cases it is handling and where it will fail. Also, lets add a todo that the validation check needs to be improved.

Reply:
ack